### PR TITLE
feat: bumping boss room to 1.1

### PR DESCRIPTION
**Pull request by SamuelBellomo**
### Description

Update BR to NGO 1.1.0 and following latest good practices with new available APIs.

### Changes with NGO 1.1.0
 - [x] use managed types for netvars? For Network GUID?
 - [x] Cleanup Relay and UTP setup? see [PR that simplifies setup flow](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/2235#issuecomment-1268627773)
 - [x] Cleanup MaxSendQueueSize value from serialized as it's been removed?
 - [x] Revert default value for max payload size
 - [x] Set connection approval timeout higher, 1 sec is pretty short. If there's a packet drop, some hangups on the network, clients would get timedout too easily.
### Contribution checklist
 - n/a Tests have been added for boss room and/or utilities pack
 - [x] Release notes have been added to the [project changelog](../CHANGELOG.md) file and/or [package changelog](../Packages/com.unity.multiplayer.samples.coop/CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - n/a JIRA ticket ID is in the PR title or at least one commit message
 - n/a Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

File: Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs [hunk 1]
```diff
@@ -1,69 +0,0 @@
-using System;
```
Comment (SamuelBellomo) [inline Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs:1]: note: this is a breaking change, we should keep this in our package.

Comment (SamuelBellomo) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs:1]: and add tests with stubbed relay

Comment (SamuelBellomo) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs:1]: add tests for stubbed lobby as well?

File: Packages/com.unity.multiplayer.samples.coop/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
   "dependencies": {
     "com.unity.learn.iet-framework": "1.2.1",
     "com.unity.multiplayer.tools": "1.0.0",
-    "com.unity.netcode.gameobjects": "1.0.2",
+    "com.unity.netcode.gameobjects": "1.1.0",
```
Comment (SamuelBellomo) [inline Packages/com.unity.multiplayer.samples.coop/package.json:12]: Adding note here to not merge until we have manifest and package.json updates done. (done once NGO 1.1.0 is out)

**Review by simon-lemay-unity — DISMISSED** [review]

File: Assets/Scripts/ConnectionManagement/ConnectionState/ClientConnectingState.cs [hunk 3]
```diff
@@ -1,9 +1,12 @@
 using System;
 using System.Threading.Tasks;
+using ConnectionManagement.ConnectionState;
 using Unity.BossRoom.UnityServices.Lobbies;
 using Unity.Multiplayer.Samples.BossRoom;
 using Unity.Multiplayer.Samples.Utilities;
 using Unity.Netcode.Transports.UTP;
+using Unity.Networking.Transport.Relay;
+using Unity.Services.Relay;
```
Comment (LPLafontaineB) [inline Assets/Scripts/ConnectionManagement/ConnectionState/ClientConnectingState.cs:9]: Some of these usings can be removed

File: Assets/Scripts/ConnectionManagement/ConnectionState/OfflineState.cs [hunk 4]
```diff
@@ -1,4 +1,5 @@
 using System;
+using ConnectionManagement.ConnectionState;
 using Unity.BossRoom.ConnectionManagement;
 using Unity.BossRoom.UnityServices.Lobbies;
 using Unity.BossRoom.Utils;
```
Comment (LPLafontaineB) [inline Assets/Scripts/ConnectionManagement/ConnectionState/OfflineState.cs:5]: same here, Auth, UTP and Services.Core can be removed

File: Assets/Scripts/ConnectionManagement/ConnectionState/StartingHostState.cs [hunk 5]
```diff
@@ -1,9 +1,13 @@
 using System;
+using ConnectionManagement.ConnectionState;
 using Unity.BossRoom.Infrastructure;
 using Unity.BossRoom.UnityServices.Lobbies;
 using Unity.Multiplayer.Samples.BossRoom;
 using Unity.Netcode;
 using Unity.Netcode.Transports.UTP;
+using Unity.Networking.Transport.Relay;
+using Unity.Services.Relay;
+using Unity.Services.Relay.Models;
```
Comment (LPLafontaineB) [inline Assets/Scripts/ConnectionManagement/ConnectionState/StartingHostState.cs:10]: Same here, those three new usings are not necessary and UTP can be removed now as well

File: Assets/Scripts/ConnectionManagement/ConnectionMethod.cs [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+using Unity.BossRoom.Utils;
+using Unity.Netcode.Transports.UTP;
+using Unity.Networking.Transport.Relay;
+using Unity.Services.Authentication;
+using Unity.Services.Core;
+using Unity.Services.Relay;
+using Unity.Services.Relay.Models;
+using UnityEngine;
+
+namespace ConnectionManagement.ConnectionState
```
Comment (LPLafontaineB) [inline Assets/Scripts/ConnectionManagement/ConnectionMethod.cs:14]:
```suggestion
namespace Unity.BossRoom.ConnectionManagement.ConnectionState
```

**Review by LPLafontaineB — COMMENTED** [review]
I really like the addition of ConnectionMethod, makes everything easier to follow. Apart from nitpicks, lgtm

**Review by LPLafontaineB — DISMISSED** [review]

**Review by LPLafontaineB — APPROVED** [review]

**Review by ashwinimurt — DISMISSED** [review]

**Review by pdeschain — DISMISSED** [review]

**Review by LPLafontaineB — APPROVED** [review]

**Pull request merged**